create_derivatives/routines.py

```diff
@@ -129,9 +133,133 @@ def ocr_pdf(self):
 
 
 class ManifestMaker:
-    # TO DO: make manifests
+    """Creates a IIIF presentation manifest version 3 from JP2 files.
+
+    Creates manifest directory in bag's data directory and then creates manifest.
+
+    Returns:
+        A tuple containing human-readable message along with list of bag identifiers.
+        Exceptions are raised for errors along the way.
+    """
+
+    def __init__(self):
+        server_url = settings.IMAGESERVER_URL
+        self.resource_url = "{}/iiif/3/".format(server_url)
+        self.fac = ManifestFactory()
+        self.fac.set_base_prezi_uri("{}/manifests/".format(server_url))
+        self.fac.set_base_image_uri(self.resource_url)
+        self.fac.set_debug(settings.PREZI_DEBUG)
+        self.upgrader = Upgrader()
+        self.manifest_dir = None
+        self.jp2_path = None
+        self.jp2_files = None
+
     def run(self):
-        pass
+        bags_with_manifests = []
+        for bag in Bag.objects.filter(process_status=Bag.PDF):
+            self.bag = bag
```

Comment:
You don't need this either.

Reply:
(the self.bag variable, that is)

Reply:
Ah, yeah, I missed removing that from when I was messing with making a bag object. Thanks!